src/C2_economic_functions.py

```diff
@@ -76,74 +76,148 @@ def crf(project_life, discount_factor):
+
+    asset_lifetime: int
+        Lifetime of an asset
+
+    first_time_investment: float
+        Investment cost of an asset to be installed
+
     discount_factor: float
-        weighted average cost of capital, which is the after-tax average cost of various capital sources
-    tax: float
-        compulsory financial charge paid to the government
+        Discount factor of a project
 
     Returns
     -------
-    capex: float
-        Capital expenditure for an asset over project lifetime
-
-    Notes
-    -----
-    Tested with
-    - test_capex_from_investment_lifetime_equals_project_life()
-    - test_capex_from_investment_lifetime_smaller_than_project_life()
-    - test_capex_from_investment_lifetime_bigger_than_project_life()
+    Per-unit replacement costs of an asset. If age_asset == 0, they need to be added to the lifetime_specific_costs of the asset.
+    If age_asset != 0, it will be needed to calculate the future investment costs of a previously installed asset.
     """
 
-    # [quantity, investment, installation, weight, lifetime, om, first_investment]
-    if project_life == lifetime:
+    # Calculate number of investments
+    if project_lifetime + age_of_asset == asset_lifetime:
         number_of_investments = 1
     else:
-        number_of_investments = int(round(project_life / lifetime + 0.5))
-    # costs with quantity and import tax at t=0
-    first_time_investment = investment_t0 * (1 + tax)
-
-    for count_of_replacements in range(0, number_of_investments):
-        # Very first investment is in year 0
-        if count_of_replacements == 0:
-            capex = first_time_investment
-        else:
-            # replacements taking place in year = number_of_replacement * lifetime
-            if count_of_replacements * lifetime != project_life:
-                capex = capex + first_time_investment / (
-                    (1 + discount_factor) ** (count_of_replacements * lifetime)
-                )
-
-    # Subtraction of component value at end of life with last replacement (= number_of_investments - 1)
-    if number_of_investments * lifetime > project_life:
-        last_investment = first_time_investment / (
-            (1 + discount_factor) ** ((number_of_investments - 1) * lifetime)
+        number_of_investments = int(
+            round((project_lifetime + age_of_asset) / asset_lifetime + 0.5)
         )
+
+    replacement_costs = 0
+
+    # In case the asset_lifetime is larger then
+    latest_investment = first_time_investment
+    # Starting from first investment (in the past for installed capacities)
+    year = -age_of_asset
+
+    present_value_of_capital_expenditures = pd.Series([latest_investment], index=[year])
+
+    # Looping over replacements, excluding first_time_investment in year (0 - age_of_asset)
+    for count_of_replacements in range(1, number_of_investments):
+        # replacements taking place after an asset ends its lifetime
+        year += asset_lifetime
+
```

Comment:
you might want to add an `assert year >=0` here so an error is raised if it is not the case

Reply:
At this point, year is allowed to be lower than 0, actually, What would be important is that `abs(year)<=asset_lifetime`

Fix in #525, fbf35460aa3f8e316a87df6db4adf91a40c64718